tutorials/pyroot/pyroot002_pythonizationDecorator.py

```diff
@@ -0,0 +1,141 @@
+## \file
+## \ingroup tutorial_pyroot
+## \notebook -nodraw
+## This tutorial shows how to use the @pythonization decorator to add extra
+## behaviour to C++ user classes that are used from Python via PyROOT.
+##
+## \macro_code
+## \macro_output
+##
+## \date November 2021
+## \author Enric Tejedor
+
+import ROOT
+from ROOT import pythonization
+
+# Let's first define a new C++ class. In this tutorial, we will see how we can
+# "pythonize" this class, i.e. how we can add some extra behaviour to it to
+# make it more pythonic or easier to use from Python.
+#
+# Note: In this example, the class is defined dynamically for demonstration
+# purposes, but it could also be a C++ class defined in some library or header.
+# For more information about loading C++ user code to be used from Python with
+# PyROOT, please see:
+# https://root.cern.ch/manual/python/#loading-user-libraries-and-just-in-time-compilation-jitting
+ROOT.gInterpreter.Declare('''
+class MyClass {};
+''')
+
+# Next, we define a pythonizor function: the function that will be responsible
+# for injecting new behaviour in our C++ class `MyClass`.
+#
+# To convert a given Python function into a pythonizor, we need to decorate it
+# with the @pythonization decorator. Such decorator allows us to define which
+# is our target class, i.e. which class we want to pythonize.
+#
+# On the other hand, the decorated function - the pythonizor - must accept
+# either one or two parameters:
+# 1. The class to be pythonized (proxy object where new behaviour can be
+# injected)
+# 2. The fully-qualified name of that class (optional).
+#
+# Let's see all this with a simple example. Suppose I would like to define how
+# `MyClass` objects are represented as a string in Python (i.e. what would be
+# shown when I print that object). For that purpose, I can define the following
+# pythonizor function. There are two important things to be noted here:
+# - The @pythonization decorator has one argument that specifies our target
```

Comment:
```suggestion
# - the @pythonization decorator has one argument that specifies our target
```

Reply:
I usually do capital for bullet points, I think I'll leave it as it is.